pkg/api/types.go

```diff
@@ -44,6 +44,7 @@ type Properties struct {
 	AgentPoolProfiles       []AgentPoolProfile      `json:"agentPoolProfiles"`
 	LinuxProfile            LinuxProfile            `json:"linuxProfile"`
 	WindowsProfile          WindowsProfile          `json:"windowsProfile"`
+	ExtensionsProfile       []ExtensionProfile      `json:"extensionsProfile"`
```

Comment:
@RobBagby, I am starting to review this PR and also read your mail (thanks for the detailed description). However, I am still not convinced that the extension profile should be at the root level without describing what its targeting. 

The current approach certainly provides flexibility but at the same time makes the model of the cluster very implicit & harder to read at a glance. Also, this modeling diverges from the principle with which ACS API was designed of being very explicit of the cluster architecture. 

E.g. the fact the extension will only run on the first master is a design detail of the extension (and dependent on extension author's design choice). However, the cluster creator can still choose to say that they want to run this extension only on the master pool. And the way to describe this in the model is to either have extension profile at the root level (like it is today) refer explicitly to its target pools or each agent/master pool supports extension profile. 

Having extension profile at the root level provides advantage of targeting more than one pool at a time.

Thoughts?